pypeit/spectrographs/spectrograph.py

```diff
@@ -533,7 +538,28 @@ def get_meta_value(self, inp, meta_key, required=False, ignore_bad_header=False,
             except (KeyError, TypeError):
                 value = None
 
-        if value is None:
+
+
+        # JFH Added this bit of code to deal with situations where the header card is there but the wrong type, e.g.
+        # MJD-OBS = 'null'
+        try:
+            if self.meta_data_model[meta_key]['dtype'] == str:
+                retvalue = str(value).strip()
+            elif self.meta_data_model[meta_key]['dtype'] == int:
+                retvalue = int(value)
+            elif self.meta_data_model[meta_key]['dtype'] == float:
+                retvalue = float(value)
+            elif self.meta_data_model[meta_key]['dtype'] == tuple:
+                assert isinstance(value, tuple)
+                retvalue = value
+            castable = True
+        except:
+            retvalue = None
+            castable = False
+
+        # JFH Added the typing to prevent a crash below when the header value exists, but is the wrong type. This
```

Comment:
Are you over-riding the data model?  If so, why?

I note we have issues with `ra` and `dec` because of this which I'm 
dealing with in a separate branch.

Reply:
You need to be more conscious of writing code that does not crash just because a header card is not in the form you want it to be. This happens all the time with calibrations, and crashing is simply not an option. I realize this is a slippery slope and/or dangerous, but we can just print out warnings.